db/blip_sync_context.go

```diff
@@ -115,6 +115,8 @@ type BlipSyncContext struct {
 	readOnly bool
 
 	collections *blipCollections // all collections handled by blipSyncContext, implicit or via GetCollections
+
+	pendingInsertions map[string]struct{} // DocIDs from handleProposeChanges that aren't in the db
```

Comment:
I think this needs to be in `blipSyncCollectionContext` because the docIDs can overlap for multiple collections.